### PR TITLE
Remove implicit dependency on wslvar binary (#198)

### LINKED ISSUE #198: Error launching chrome from WSL
I'm getting an error when trying to use open@7.3.0 in WSL in a project built with @open-wc, @web/dev-server

**This error doesn't happen with open@7.2.1.**

**Steps to reproduce:**

1. Create a new project using:

`npm init @open-wc --type scaffold --scaffoldType wc --features linting testing demoing --typescript true --scaffoldFilesFor testing demoing --tagName ui-element --writeToDisk true --installDependencies npm
cd ui-element
`
2. run with npm start

The following error is displayed.  

[web-dev-server] Error: spawn wslvar ENOENT
[web-dev-server]     at Process.ChildProcess._handle.onexit (internal/child_process.js:267:19)
[web-dev-server]     at onErrorNT (internal/child_process.js:469:16)
[web-dev-server]     at processTicksAndRejections (internal/process/task_queues.js:84:21) {   
[web-dev-server]   errno: 'ENOENT',
[web-dev-server]   code: 'ENOENT',
[web-dev-server]   syscall: 'spawn wslvar',
[web-dev-server]   path: 'wslvar',
[web-dev-server]   spawnargs: [ 'systemroot' ],
[web-dev-server]   cmd: 'wslvar systemroot',
[web-dev-server]   stdout: '',
[web-dev-server]   stderr: ''
[web-dev-server] }

### PR DESCRIPTION
**Pull request by tim-stasse**
Fixes #198
Fixes #200

### CODE REVIEW

**Comment by tim-stasse** [comment]
I believe this would make [200](https://github.com/sindresorhus/open/pull/200) redundant (if accepted).

File: index.js [hunk 1]
```diff
@@ -19,16 +20,47 @@ const wslToWindowsPath = async path => {
...
 
-// Convert a path from Windows format to WSL format
-const windowsToWslPath = async path => {
-	const {stdout} = await pExecFile('wslpath', [path]);
-	return stdout.trim();
+// Get the directory where fixed drives are automatically mounted in WSL
+const wslAutoMountRoot = async () => {
+	let rootDirectory = '/mnt/';
+	try {
+		const {stdout} = await pExecFile('awk', [
```
Comment (sindresorhus) [inline index.js:27]: Use JS for the parsing instead of `awk`.

File: index.js [hunk 2]
```diff
@@ -19,16 +20,47 @@ const wslToWindowsPath = async path => {
...
+	const rootDirectory = await wslAutoMountRoot();
+	let systemDrive = 'c';
+	const systemDirectory = '/Windows/System32';
+
+	try {
+		// List the contents of the root directory.
+		const paths = await pReaddir(rootDirectory);
+
+		/* eslint-disable no-await-in-loop */
+		for (const path of paths) {
```
Comment (sindresorhus) [inline index.js:55]: This can be done concurrently using `Promise.all`.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
Can you describe your changes and why they're required?

**Review by jonaskuske — CHANGES_REQUESTED** [review]
Since this already adds a detection for `wslAutoMountRoot`, you could use that as well for the check on line 125 instead of hardcoding `/mnt/` – that'd make my PR (#210) unnecessary :)

**Comment by sindresorhus** [comment]
@tim-stasse Friendly bump :)

**Comment by anaisbetts** [comment]
@sindresorhus I think I can explain - `wslvar` is not a built-in component, it's optional software that many people don't have. While Powershell is guaranteed to exist, `wslvar` is very much not. fwiw, this bug breaks popular software such as Storybook on WSL because open fails

**Pull request closed without merging**